overlord/snapstate/refreshhints.go

```diff
@@ -121,3 +135,111 @@ func (r *refreshHints) Ensure() error {
 	}
 	return r.refresh()
 }
+
+func refreshHintsFromCandidates(st *state.State, updates []*snap.Info, ignoreValidationByInstanceName map[string]bool, deviceCtx DeviceContext) ([]*RefreshHint, error) {
+	if ValidateRefreshes != nil && len(updates) != 0 {
+		userID := 0
+		var err error
+		updates, err = ValidateRefreshes(st, updates, ignoreValidationByInstanceName, userID, deviceCtx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	hints := make([]*RefreshHint, len(updates))
+	for _, update := range updates {
+		var snapst SnapState
+		if err := Get(st, update.InstanceName(), &snapst); err != nil {
+			return nil, err
+		}
+		// XXX: is this ok?
```

Comment:
what worries you here?

Reply:
It's just about semantics of flags and where they come from, it's mostly a note to myself to double-check this.